.circleci/config.yml

```diff
@@ -0,0 +1,120 @@
+
+commands:
+  check-for-changes:
+    description: "Searches for any files matching the pattern in a diff result"
+    parameters:
+      in:
+        description: "Returns true if this pattern matches any folder/file in the diff"
+        type: string
+      since:
+        description: "The git commit that we are checking against"
+        type: string
+    steps:
+      - run:
+          command: |
+            git diff --name-only HEAD << parameters.since >> | grep << parameters.in >>
+
+  write-build-info:
+    description: "Write relevant info to the current build/deploy to a file for referencing"
+    parameters:
+      data:
+        description: "The contents of the build file"
+        type: string
+        default: |
+          BUILD_COMMIT=<< pipeline.git.revision >>
+      dest:
+        description: "The destination of the build file"
+        type: string
+    steps:
+      - run:
+          name: "Writing build info"
+          command: |
+            echo "<< parameters.data >>" > << parameters.dest >>/BUILD.txt
+
+  copy-to-s3:
+    description: "Copies adirectory to a bucket (and folder) in S3"
+    parameters:
+      source:
+        description: "The local directory to copy"
+        type: string
+      bucket:
+        description: "The S3 bucket (and optional folder) to copy everything to"
+        type: string
+    steps:
+      - run: aws s3 cp << parameters.source >> s3://<< parameters.bucket >> --recursive --acl public-read
+
+jobs:
+  needs-to-publish-storybook:
+    docker:
+      - image: circleci/node:14.5.0
+    parameters:
+      doc_site:
+        type: string
+        description: "The domain for the doc site containing the build file"
+    steps:
+      - checkout
+      - run:
+          name: "Get last published commit"
+          command: |
+            echo "export LAST_PUBLISHED_COMMIT=$(curl -s -f https://<< parameters.doc_site >>/storybook/BUILD.txt | sed -r 's/^.*=(.*)$/\1/')" >> $BASH_ENV
+            source $BASH_ENV
+        # If we didn't find a last published commit, then we need to publish
+      - run: |
+          if [ "$LAST_PUBLISHED_COMMIT" = "" ]; then
+            circleci-agent step halt
+          fi
+      - check-for-changes:
+          in: "formation-react"
+          since: $LAST_PUBLISHED_COMMIT
+
+  publish-storybook:
+    executor: aws-cli/default
+    environment:
+      AWS_DEFAULT_REGION: us-gov-west-1
+    parameters:
+      bucket:
+        type: string
+        description: "The S3 bucket to copy the storybook build into"
+    steps:
+      - checkout
+      - aws-cli/setup
+      - run: yarn install
+      - run:
+          name: "Build Storybook"
+          command: cd packages/formation-react && yarn build-storybook
+      - write-build-info:
+          dest: "packages/formation-react/storybook-static"
+      - copy-to-s3:
+          source: "./packages/formation-react/storybook-static"
+          bucket: "<< parameters.bucket >>/storybook"
```

Comment:
This job will run in the workflow if we've determined that we need to publish storybook. It uses the [CircleCI `aws-cli` orb](https://circleci.com/developer/orbs/orb/circleci/aws-cli), and the `context` in the workflow is set up with environment variables that the orb can use for the key & secret key.

This does a bit of initial setup, then the meat of the job is:

1. Build storybook
1. Write a `BUILD.txt` file to the storybook static site directory
1. Copy all of the files in the storybook static site directory to the S3 bucket, making sure that they are publicly readable.